bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/types/TypeFlags.java

```diff
@@ -53,4 +53,18 @@ public static int addToMask(int mask, int... flags) {
         }
         return mask;
     }
+
+    public static int getAnydataTypeFlag(BType type) {
```

Comment:
Still not sure if this belongs here. IMO, we rarely create a type at runtime. Since the type creation logic is iterator specific, shall we move this somewhere else? We can maybe combine both methods then to have a single method that does both checks.

Reply:
Added a new jvm class IteratorUtils and added thos two methods